test/Streamly/Test/FileSystem/Event.hs

```diff
@@ -181,19 +184,28 @@ driver ::
        , FilePath -> IO ()
        , FilePath -> IO ()
        , [String]
+       , Bool
        )
     -> SpecWith ()
-driver ec (desc, pre, ops, events) = it desc $ runTest `shouldReturn` "PASS"
+driver ec (desc, pre, ops, events, sym) = it desc $ runTest `shouldReturn` "PASS"
 
     where
 
     runTest = do
         sync <- newEmptyMVar
-        withSystemTempDirectory fseventDir $ \fp -> do
-            pre fp
-            let eventStream = checker ec fp sync events
-                fsOps = S.fromEffect $ runFSOps fp sync
-            fmap fromJust $ S.head $ eventStream `S.parallelFst` fsOps
+        withSystemTempDirectory fseventDir $ \fp ->
+            if sym
+            then do
+                  createDirectoryLink fp (fp  ++ "SymLink")
+                  pre (fp  ++ "SymLink")
+                  let eventStream = checker ec (fp  ++ "SymLink") sync events
+                      fsOps = S.fromEffect $ runFSOps (fp  ++ "SymLink") sync
+                  fmap fromJust $ S.head $ eventStream `S.parallelFst` fsOps
+            else do
+                  pre fp
+                  let eventStream = checker ec fp sync events
+                      fsOps = S.fromEffect $ runFSOps fp sync
+                  fmap fromJust $ S.head $ eventStream `S.parallelFst` fsOps
```

Comment:
The code in the "else" part is common with the "if" branch, you can just pass `fp` or `fp ++ "SymLink".`